db/migrations/20210111171826_create_licences.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const tableName = 'licences'
+
+exports.up = async function (knex) {
+  await knex
+    .schema
+    .createTable(tableName, table => {
+      // Primary Key
+      table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'))
+
+      // Data
+      table.uuid('invoice_id').notNullable()
+      table.string('line_attr1').notNullable() // Licence number is represented as line_attr1
+      table.string('customer_reference').notNullable()
+      table.integer('financial_year').notNullable()
+
+      table.integer('credit_count').notNullable().defaultTo(0)
+      table.bigInteger('credit_value').notNullable().defaultTo(0)
+
+      table.integer('debit_count').notNullable().defaultTo(0)
+      table.bigInteger('debit_value').notNullable().defaultTo(0)
+
+      table.integer('zero_count').notNullable().defaultTo(0)
+
+      table.integer('new_licence_count').notNullable().defaultTo(0)
+
+      // There can only be 1 customer summary per invoice per financial year for a licence
+      table.unique(['invoice_id', 'line_attr1', 'customer_reference', 'financial_year'])
```

Comment:
Each combination of bill run ID, regime, customer reference and financial year will give us a unique invoice ID.

Because of this, we don't have to keep track of customer reference or financial year in the `licence` or use it in a key. So you could drop those 2 columns and just use

```suggestion
      table.unique(['invoice_id', 'line_attr1'])
```

Though bear in mind my previous comment about using a proper name for the column which holds the licence number.